### ISSUE TITLE
Logic for multiple reload jobs merging into a running reload is borked

### ISSUE THREAD

**Issue by bl33pbl0p** [systemd/systemd#11499]
**systemd version the issue has been seen with**
> 240

**Used distribution**
 > Debian

**Expected behaviour you didn't see**
 > Intended semantics seem to be that triggering multiple reload requests when a reload job is in progress (running) should actually execute the ExecReload= thing as many times (or atleast once more, after it completes).  As per https://github.com/systemd/systemd/issues/10896#issuecomment-442191551

**Unexpected behaviour you saw**
 > Triggering another reload request causes reload job to get merged into the installed reload job, but never to be re-run.

**Steps to reproduce the problem**
```ini
[Service]
ExecStart=/bin/sleep infinity
ExecReload=/bin/sleep 10
```

```bash
% systemctl reload foo --no-block; sleep 5; systemctl reload foo --no-block
```

Service is reloaded only once. Also, when the second reload job is triggered, the installed job instead finishes, result=done, and nothing shows in systemctl list-jobs (as it completes). Service state on reception of SIGCHLD for spawned child is changed back to running from reloading.

**Comment by bl33pbl0p** [comment]
I figured it out.

The first reload job goes on as usual, the second reload job however is merged into the installed job, and set to JOB_WAITING. Then, when the manager dispatches run_queue, it calls job_perform_on_unit, which goes through unit_reload->unit_active_state->service_active_state which is reloading, so EALREADY is returned, and the job is finished with JOB_DONE result, already=true, and no reload payload is spawned again.

Fix is to probably return EAGAIN instead, so that it keeps waiting until that reload operation is compelete. Since other requests until it is in run_queue keep getting merged into installed job, it would all coalesce into a single waiting job and be executed only once after the previous binary finishes executing.

**Comment by bl33pbl0p** [comment]
Breakpoint 152, manager_dispatch_run_queue (source=0x5568c4178d20, userdata=0x5568c4179190) at ../src/core/manager.c:2131
2131	        Manager *m = userdata;
(gdb) 
Continuing.

Breakpoint 24, job_run_and_invalidate (j=0x5568c4243a80) at ../src/core/job.c:664
664	        assert(j);
(gdb) 
Continuing.

Breakpoint 47, job_is_runnable (j=0x5568c4243a80) at ../src/core/job.c:452
452	static bool job_is_runnable(Job *j) {
(gdb) 
Continuing.

Breakpoint 27, job_start_timer (j=0x5568c4243a80, job_running=true) at ../src/core/job.c:1084
1084	int job_start_timer(Job *j, bool job_running) {
(gdb) 
Continuing.

Breakpoint 54, job_set_state (j=0x5568c4243a80, state=JOB_RUNNING) at ../src/core/job.c:109
109	        assert(j);
(gdb) 
Continuing.

Breakpoint 1, job_add_to_dbus_queue (j=0x5568c4243a80) at ../src/core/job.c:1153
1153	        assert(j);
(gdb) 
Continuing.

Breakpoint 51, job_perform_on_unit (j=0x7ffe2194e3e8) at ../src/core/job.c:623
623	        assert(j);
(gdb) 
Continuing.

Breakpoint 225, service_can_reload (u=0x5568c4242790) at ../src/core/service.c:2424
2424	        Service *s = SERVICE(u);
(gdb) 
Continuing.

Breakpoint 216, service_active_state (u=0x5568c4242790) at ../src/core/service.c:2937
2937	        assert(u);
(gdb) 
Continuing.

Breakpoint 77, manager_get_job (m=0x5568c4179190, id=258) at ../src/core/manager.c:1854
1854	        assert(m);
(gdb) 
Continuing.

Breakpoint 10, job_finish_and_invalidate (j=0x5568c4243a80, result=JOB_DONE, recursive=true, already=true) at ../src/core/job.c:977
977	int job_finish_and_invalidate(Job *j, JobResult result, bool recursive, bool already) {
(gdb) 
Continuing.

Breakpoint 23, job_result_to_string (i=JOB_DONE) at ../src/core/job.c:1654
1654	DEFINE_STRING_TABLE_LOOKUP(job_result, JobResult);
(gdb) 
Continuing.

Breakpoint 36, job_type_to_string (i=JOB_RELOAD) at ../src/core/job.c:1626
1626	DEFINE_STRING_TABLE_LOOKUP(job_type, JobType);
(gdb) 
Continuing.

Breakpoint 37, job_uninstall (j=0x5568c4243a80) at ../src/core/job.c:137
137	        assert(j->installed);
(gdb) 
Continuing.

br for unit.c is missing but you get the idea
